packages/rpc-provider/src/coder/index.ts

```diff
@@ -68,8 +70,9 @@ export class RpcCoder {
   private _checkError (error?: JsonRpcResponseBaseError): void {
     if (error) {
       const { code, data, message } = error;
+      const fmtMessage = `${code}: ${message}${formatErrorData(data)}`;
```

Comment:
Does it make sense to keep the formatted message? If we're making the other fields available separately, maybe we shouldn't change the message.

Reply:
I prefer the message formatted. This maintains full compatibility, ie. if you don’t yet use the new fields you have exactly the same information you had previously.